### PR TITLE
Add useGa hook for recording GA events

### PR DESCRIPTION
**Pull request by flozia**
# Description

In PR #2045 it was [pointed out](https://github.com/mozilla/blurts-server/pull/3402#discussion_r1326956619) that we’d like to refrain from using the current analytics script we use in prod for the new site. This is adding a useGtag hook for recording GA events.

### CODE REVIEW

**Review by Vinnl — APPROVED** [review]
Thanks for adding this. Are there any events that should be sent to Glean but not to GA, or the other way around? Because if not, we `useAnalytics` wrapper that sends it to both would be a nice addition, rather than having callers load both hooks.

File: src/app/hooks/useAnalytics.ts
Comment (flozia) [reply src/app/hooks/useAnalytics.ts]: Rephrased in [b2c287c](https://github.com/mozilla/blurts-server/pull/3440/commits/b2c287c6b65960544eb605a6e754e496d705ac04).

File: src/app/hooks/useAnalytics.ts
Comment (flozia) [reply src/app/hooks/useAnalytics.ts]: That’s true. Might be to fine-grained to also log this — moved the log [b2c287c](https://github.com/mozilla/blurts-server/pull/3440/commits/b2c287c6b65960544eb605a6e754e496d705ac04).

File: src/app/(proper_react)/layout.tsx
Comment (flozia) [reply src/app/(proper_react)/layout.tsx]: Definitely — thanks for the suggestion: [8833c94](https://github.com/mozilla/blurts-server/pull/3440/commits/8833c9439ebdc616de1434ce44594940e52e65b6).

Comment (flozia) [reply src/app/(proper_react)/layout.tsx]: Also retrieving `nonce` inline sharing the hopes of not having to add another inline script: [dd19556](https://github.com/mozilla/blurts-server/pull/3440/commits/dd195565973e519b1528181d1b5b50c90d8e9e69).

**Comment by flozia** [comment]
> Thanks for adding this. Are there any events that should be sent to Glean but not to GA, or the other way around? Because if not, we `useAnalytics` wrapper that sends it to both would be a nice addition, rather than having callers load both hooks.

I’d like to have a wrapper in the future that would enable us to report to both and I think this would be a good addition. I did not add this yes because there might be slightly different events/campaigns we are planning to get metrics on with GA and Glean — if I understood correctly @rhelmer? I created [MNTOR-2195](https://mozilla-hub.atlassian.net/browse/MNTOR-2195) to not lose track of the proposal. Also, we’ll need to go through an approval process for every Glean metric we’d like to add and before starting to report to both I’d like to have more clarity internally on what we’ll utilize each platform for.

I’ll go ahead with emerging this since it will unblock PR https://github.com/mozilla/blurts-server/pull/3402.

**Pull request merged**
